client/logclient_test.go

```diff
@@ -712,3 +721,69 @@ func TestGetAcceptedRootsErrors(t *testing.T) {
 		}
 	}
 }
+
+func TestGetEntryAndProof(t *testing.T) {
+	hs := serveRspAt(t, "/ct/v1/get-entry-and-proof", GetEntryAndProofResp)
+	defer hs.Close()
+	lc, err := client.New(hs.URL, &http.Client{}, jsonclient.Options{})
+	if err != nil {
+		t.Fatalf("Failed to create client: %v", err)
+	}
+
+	tests := []struct {
+		index    uint64
+		treesize uint64
+	}{
+		{1000, 2000},
+	}
+
+	for _, test := range tests {
+		resp, err := lc.GetEntryAndProof(context.Background(), test.index, test.treesize)
+		if err != nil {
+			t.Errorf("GetEntryAndProof(%v, %v)=nil,%v; want proof,nil", test.index, test.treesize, err)
+		} else if got := len(resp.AuditPath); got < 1 {
+			t.Errorf("len(GetEntryAndProof(%v, %v)): %v; want > 1", test.index, test.treesize, got)
+		}
+	}
+}
+
+func TestGetEntryAndProofErrors(t *testing.T) {
+	ctx := context.Background()
+	var tests = []struct {
+		rsp, want string
+	}{
+		{rsp: "", want: "EOF"},
+		{rsp: "not-json", want: "invalid"},
+		{rsp: `{"leaf_input": "bogus", "extra_data": "Z29vZAo=", "audit_path": ["Z29vZAo="]}`, want: "illegal base64"},
+		{rsp: `{"leaf_input": "Z29vZAo=", "extra_data": "bogus", "audit_path": ["Z29vZAo="]}`, want: "illegal base64"},
+		{rsp: `{"leaf_input": "Z29vZAo=", "extra_data": "Z29vZAo=", "audit_path": ["bogus"]}`, want: "illegal base64"},
+		{rsp: `{"leaf_input": "Z29vZAo=", "extra_data": "Z29vZAo=", "audit_path": ["bbbb",]}`, want: "invalid"},
+	}
+
+	for _, test := range tests {
+		ts := serveRspAt(t, "/ct/v1/get-entry-and-proof", test.rsp)
```

Comment:
Maybe move this and `client.New` to above the loop?

Reply:
Cut and pasted.

Reply:
Did?

Reply:
No I left it the same as the other tests for the moment.